xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/UserWikiSheet.xml

```diff
@@ -468,4 +468,148 @@ return XWiki;
+        <editor>Text</editor>
+        <name>content</name>
+        <number>1</number>
+        <prettyName>Executed Content</prettyName>
+        <rows>25</rows>
+        <size>120</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </content>
+      <extensionPointId>
+        <disabled>0</disabled>
+        <name>extensionPointId</name>
+        <number>5</number>
+        <prettyName>Extension Point ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </extensionPointId>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>6</number>
+        <prettyName>Extension ID</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parameters>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>parameters</name>
+        <number>7</number>
+        <prettyName>Extension Parameters</prettyName>
+        <rows>10</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </parameters>
+      <scope>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>scope</name>
+        <number>8</number>
+        <prettyName>Extension Scope</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>wiki=Current Wiki|user=Current User|global=Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </scope>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <content/>
+    </property>
+    <property>
+      <extensionPointId>org.xwiki.plaftorm.user.profile.menu.content</extensionPointId>
+    </property>
+    <property>
+      <name>org.xwiki.platform.user.profile.entry.userwikisheet</name>
+    </property>
+    <property>
+      <!-- display: The current wiki is the main wiki and the user sheet document exists.-->
+      <parameters>id=wikis
+name=$services.localization.render('platform.wiki.menu.userwikis')
+sheet=WikiManager.UserWikiSheet
+glyphicon=list
+display=#if($xcontext.isMainWiki() &amp;&amp; $xwiki.exists('WikiManager.UserWikiSheet'))true#{else}false#end</parameters>
```

Comment:
I don't understand the check for ` $xwiki.exists('WikiManager.UserWikiSheet'))`. BTW your XML comment above is not great since it explains the code (which is almost never useful since we can read it ;)). A good comment must explain WHY it's done.

Reply:
If the sheet doesn't exist then this UIX doesn't exist.

Reply:
> I don't understand the check for $xwiki.exists('WikiManager.UserWikiSheet')). 

Sounds like a too quick copy/pasted from the old version. The good thing with UIX is that you don't need this kind of hack anymore.

Reply:
I've removed the useless condition.